### PR TITLE
feat: add more information in spans

### PR DESCRIPTION
**Pull request by bnjjj**
Related to #911

### CODE REVIEW

File: apollo-router-core/src/query_planner/mod.rs [hunk 1]
```diff
@@ -218,7 +221,11 @@ impl PlanNode {
                         )
                         .instrument(tracing::info_span!(
                             "fetch",
-                            "otel.kind" = %SpanKind::Internal
+                            "otel.kind" = %SpanKind::Client,
+                            "net.peer.name" = &field::Empty,
```
Comment (Geal) [inline apollo-router-core/src/query_planner/mod.rs:225]: do we need to add the net and http info if it's empty?

**Review by Geal — CHANGES_REQUESTED** [review]

**Review by BrynCooke — CHANGES_REQUESTED** [review]
Gut feeling is that this information should be in a span in the TowerSubgraphService.
Eventually it is very possible that we will have other types of subgraph service that do not actually make an http call.

File: apollo-router-core/src/query_planner/mod.rs (hunk 1 above)
Comment (bnjjj) [reply apollo-router-core/src/query_planner/mod.rs:225]: It's not empty, it's updated inside the span. But we always need to declare all the record we could have in a span, we can't add them dynamically, only edit them dynamically

**Comment by bnjjj** [comment]
@BrynCooke Not sure to understand your comment. At the moment I only put these records on the `fetch` span which is the function to make the http call to the subgraph. It's not on the `subgraph_service` span right now it's only used on `fetch_node` method which make the http request.

**Comment by BrynCooke** [comment]
Yeah, I think the fetch span should stay as it is, and that a new span should go in TowerSubgraphService.
It's TowerSubgraphService that actually does the 'client' work and knows all about http stuff.

Having this client span fetch means that anything in the subgraph service pipeline such as caching would also be marked as client.

I guess the point I'm trying to make is that the existing fetch span is too early to be a client span.

**Review by BrynCooke — APPROVED** [review]

**Review by Geal — APPROVED** [review]

**Pull request merged**